crates/engine/tree/src/engine.rs

```diff
@@ -0,0 +1,173 @@
+//! An engine API handler for the chain.
+
+use crate::{
+    chain::{ChainHandler, FromOrchestrator, HandlerEvent, OrchestratorState},
+    tree::EngineApiTreeHandler,
+};
+use reth_beacon_consensus::BeaconEngineMessage;
+use reth_primitives::{SealedBlockWithSenders, B256};
+use std::{
+    collections::VecDeque,
+    pin::Pin,
+    task::{Context, Poll},
+};
+
+/// Advances the chain based on incoming requests.
+///
+/// This is a general purpose request handler with network access.
+/// This type listens for incoming messages and processes them via the configured request handler.
+///
+/// ## Overview
+///
+/// This type is an orchestrator for incoming messages and responsible for delegating requests
+/// received from the CL to the handler.
+///
+/// It is responsible for handling the following:
+/// - Downloading blocks on demand from the network if requested by the [`EngineApiRequestHandler`].
+///
+/// The core logic is part of the [EngineRequestHandler], which is responsible for processing the
+/// incoming requests.
+pub struct EngineHandler<T>
+where
+    T: EngineRequestHandler,
+{
+    /// Processes requests.
+    ///
+    /// This type is responsible for processing incoming requests.
+    handler: T,
+    /// Receiver for incoming requests that need to be processed.
+    // TODO add stream type for T::Request,
+    incoming_requests: (),
```

Comment:
These receivers should be polled in a spawned task in the background right? or will the Engine Handler itself be the thing that's spawned in the background?